src/grt/grt-wave_options-parse.adb

```diff
@@ -0,0 +1,318 @@
+      end if;
+
+      --~ Debug.Dump_Tree;
+
+   end Start;
+
+-- private --------------------------------------------------------------------
+
+   procedure Parse_Version (Line : String_Acc) is
+      Msg_Invalid_Format : constant String := "invalid version format";
+      First, Dot_Index, Num : Integer;
+   begin
+
+      if Version /= (others => -1) then
+         Error_Context ("version is set more than once");
+      end if;
+
+      if Trees /= Tree_Array'(others => null) then
+         Error_Context ("version cannot be set after signal paths");
+      end if;
+
+      First := First_Non_Whitespace_Pos (Line (Line'First + 1 .. Line'Last));
+      if Line (First .. First + 6) /= "version" then
+         Error_Context (Msg_Invalid_Format);
+      end if;
+
+      First := First + 7;
+      if not Is_Whitespace (Line (First)) then --catch "version\n","version1.0"
+         Error_Context (Msg_Invalid_Format);
+      end if;
+
+      First := First_Non_Whitespace_Pos (Line (First + 1 .. Line'Last));
+      if First = -1 then -- catch "version \n", "version  \n", etc
+         Error_Context (Msg_Invalid_Format);
+      end if;
+
+      Dot_Index := Find (Line (First + 1 .. Line'Last), '.');
+      if Dot_Index = -1 then -- catch the absence of "." or "version ."
+         Error_Context (Msg_Invalid_Format);
+      end if;
+
+      Num := Value (Line (First .. Dot_Index - 1));
+      if Num = -1 then -- catch version a.2
+         Error_Context (Msg_Invalid_Format);
+      end if;
+      Version.Major := Num;
+
+      Num := Value (Line (Dot_Index + 1 .. Line'Last));
+      if Num = -1 then -- catch version 1.a
+         Error_Context (Msg_Invalid_Format);
+      end if;
+      Version.Minor := Num;
+
+      if Version.Major /= Current_Version.Major
+        or else Version.Minor > Current_Version.Minor
+      then
+         Print_Context (Error);
+         Error_C ("unsupported format version; it must be ");
+         if Current_Version.Minor /= 0 then
+            Error_C ("between ");
+            Print_Version (Version_Type'(Current_Version.Major, 0));
+            Error_C (" and ");
+         end if;
+         Print_Version (Current_Version);
+         Error_E;
+      end if;
+
+   end Parse_Version;
+
+   procedure Print_Version (Version : Version_Type) is
+      Num_Str : String (1 .. Value_String_Size);
+      First : Positive;
+   begin
+      To_String (Num_Str, First, Ghdl_I32 (Version.Major));
+      Report_C (Num_Str (First .. Num_Str'Last));
+      Report_C (".");
+      To_String (Num_Str, First, Ghdl_I32 (Version.Minor));
+      Report_C (Num_Str (First .. Num_Str'Last));
+   end Print_Version;
+
+   --------------------------------------------------------------------------
+
+   procedure Parse_Path (Line : String_Acc) is
+      First, Last : Positive;
+      Tree_Updated : Boolean;
+      Tree_Index : Tree_Index_Type;
+   begin
+      To_Lower (Line_Context.Str.all);
```

Comment:
I will have a look on how ghdl reports compilation error in order to print errors/warning in a more similar format.

Actually I need to save the line because when the user makes a typo in a path like this :
/top/sub1/sub2/sub2/sub4/a
instead of :
/top/sub1/sub2/sub2/sub4/a​​

Then he will get a warning telling him that /top/sub1/sub2/sub2 wasn't found in design. Then he will immediately now that he made a typo with subb3.
I believe that this is really helpfull because if the path is long it isn't always obvious to find the error.
Maybe, I could also just warn that sub2 wasn't found, although it could exist several times in the path (as in this example​), then it won't point exactly to the problematic one.

But I think there is a better way to do that : I can add a parent field to the elements of the tree so that it is possible to find the full path of an element because we could now go from this element to the top of the tree.